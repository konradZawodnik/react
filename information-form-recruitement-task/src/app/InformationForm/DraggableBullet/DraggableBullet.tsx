import { Button } from "@/components/ui/button";
import { Input } from "@/components/ui/input";
import { useDrag } from "react-dnd";
import { ItemType } from "../InformationForm";

export const DraggableBullet = ({
    index,
    bullet,
    moveBullet,
    handleRemoveBullet,
    handleChangeBullet,
  }: {
    index: number;
    bullet: string;
    moveBullet: (dragIndex: number, hoverIndex: number) => void;
    handleRemoveBullet: (index: number) => void;
    handleChangeBullet: (index: number, value: string) => void;
  }) => {
    const [{ isDragging }, drag] = useDrag({
      type: ItemType,
      item: { index },
      collect: (monitor) => ({
        isDragging: monitor.isDragging(),
      }),
    });
  
    return (
      <div
        ref={drag as any}
        className={`border border-gray-300 cursor-move flex mb-4 p-2 ${
          isDragging ? "opacity-50" : "opacity-100"
        }`}
      >
        <Input
          className="h-100 min-h-[40px] p-2"
          value={bullet}
          onChange={(e) => handleChangeBullet(index, e.target.value)}
          type="text"
          placeholder={`Bullet point #${index + 1}`}
        />
        <Button
          className="bg-[#ff0000] border border-solid cursor-pointer hover:bg-[#0068fa] p-2 ml-1 rounded text-white"
          variant="secondary"
          onClick={() => handleRemoveBullet(index)}
        >
          Remove
        </Button>
      </div>
    );
  };