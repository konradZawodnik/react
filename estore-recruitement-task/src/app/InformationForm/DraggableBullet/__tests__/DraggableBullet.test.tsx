import { render } from "@testing-library/react";
import { DraggableBullet } from "../DraggableBullet";
import "@testing-library/jest-dom";

global.ResizeObserver = jest.fn().mockImplementation(() => ({
  observe: jest.fn(),
  unobserve: jest.fn(),
  disconnect: jest.fn(),
}));

jest.mock("react-dnd", () => ({
  useDrag: jest
    .fn()
    .mockImplementation(() => [jest.fn(), jest.fn(), jest.fn()]),
}));

const index = 1;
const mockBullet = "testBullet";
const mockMoveBullet = jest.fn();
const mockHandleRemoveBullet = jest.fn();
const mockHandleChangeBullet = jest.fn();

describe("DraggableBullet", () => {
  it("should match snapshot", () => {
    const { container } = render(
      <DraggableBullet
        index={index}
        bullet={mockBullet}
        moveBullet={mockMoveBullet}
        handleRemoveBullet={mockHandleRemoveBullet}
        handleChangeBullet={mockHandleChangeBullet}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
