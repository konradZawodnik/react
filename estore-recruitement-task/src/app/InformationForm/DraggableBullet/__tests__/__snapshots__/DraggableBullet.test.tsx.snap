// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DraggableBullet should match snapshot 1`] = `
<div>
  <div
    class="border border-gray-300 cursor-move flex mb-4 p-2 opacity-100"
  >
    <input
      class="flex w-full rounded-md border border-input bg-transparent text-base shadow-sm transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50 md:text-sm h-100 min-h-[40px] p-2"
      placeholder="Bullet point #2"
      type="text"
      value="testBullet"
    />
    <button
      class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 shadow-sm h-9 bg-[#ff0000] border border-solid cursor-pointer hover:bg-[#0068fa] p-2 ml-1 rounded text-white"
    >
      Remove
    </button>
  </div>
</div>
`;
